sdk/storage/azblob/internal/testcommon/clients_auth.go

```diff
@@ -364,3 +368,27 @@ func GetAccountSAS(permissions sas.AccountPermissions, resourceTypes sas.Account
 
 	return sasQueryParams.Encode(), nil
 }
+
+func DeleteContainerUsingManagementClient(_require *require.Assertions, accountType TestAccountType, containerName string) {
```

Comment:
Do we need to use the management client? We could also list containers in the account with container name prefix or get container properties and see if a lease is held on the client, break lease if needed and then delete

Reply:
In this case, Immutability policy is set on the account level. So, the container deletion fails using the client created from the azblob package. This is why I am using the management client here.
Get properties on the container returns `available` in the lease state and `unlocked` in the lease status. So, the container doesn't have lease here.